### ISSUE TITLE
Books app windows seem to be ignored

### ISSUE THREAD

**Issue by imikushin** [lwouis/alt-tab-macos#627]
### Describe the bug

Book window opened by macOS built-in Books app is not shown by AltTab.

### Screenshots / video

Here's a little video to demonstrate: https://www.dropbox.com/s/s03v3otpweszgez/BooksAreInvisible.mov?dl=0
Notice that there is a book open, but AltTab doesn't see it.

### Steps to reproduce the bug

1. Open 'Books' app (the 'Library' tab will be open in Books app window)
2. Use AltTab to switch to another window and back to Books (this still works with the main Books window, _unless_ you interacted with it: open a book, navigate to Audiobooks, etc)
3. Open a book to read (in my case, 'The Swift Programming Language')
4. Use AltTab to switch to another window. **What's wrong:** AltTab doesn't show the open book window, so you can't switch back to it.

Interesting that when you close the book window, and the Books main (Library) window shows up instead, you can again see it in AltTab.

### Environment

AltTab v6.6.0 and v6.7.1
OS: macOS Catalina (10.15.6) and BigSur (11 beta 8)

**Comment by lwouis** [comment]
Hi @imikushin! Thank you for this high quality bug report!

I worked on supporting Books.app previously in #481. I learned then how the animation that takes place when opening a book is hard to handle because from a technical perspective we see a bogus window created, then only after the animation finishes, the window is a normal window from a technical perspective. This means i had to add observing so that i can test if the window is legit after the animation.

I'm sad to see that the workaround isn't working for you. Could you share a bit more about how it behaves?

Like if you close the book and go back to the library, do you always see the main window? When you open a book, is it never shown in AltTab, or just sometimes? Maybe it would help if you recorded another video playing around with Books, like opening closing windows, fullscreening them, and in-between each action, check what's show in AltTab.

**Comment by karamfd** [comment]
I'm having the same issue as @imikushin with the latest version, 6.9. I'm also using Mac OS Catalia 10.15.7

**Comment by lwouis** [comment]
@karamfd could you please share more info to help me investigate? I wrote above:

> Like if you close the book and go back to the library, do you always see the main window? When you open a book, is it never shown in AltTab, or just sometimes? Maybe it would help if you recorded another video playing around with Books, like opening closing windows, fullscreening them, and in-between each action, check what's show in AltTab.

**Comment by karamfd** [comment]
@lwouis The app doesn't work at all when the book is opened. When I close the book and go back to the Library, it doesn't work most of the time either. 

I can get it to work sometimes when I start the books app in the Library window. If I do anything else like opening a book, the app stops working altogether.

**Comment by phamousq** [comment]
+1 on this issue, Alt Tab does not see the window with text from the opened book after I open a book from the library. Version 6.12.0, Big Sur 11.1

**Comment by lwouis** [comment]
I investigated this issue more:

* When opening a book, AltTab receives event that a window is created
* AltTab correctly show the window (invoke alt-tab during the animation correctly shows the first visual of the window)
* Books.app then closes the window, and re-opens a new one
* AltTab receives the event for the window being closed, and stop listing it
* AltTab doesn't receive the event for the new window being created, so can't list it

So in short, Books.app is doing shenanigans when opening books. It's doing a complex animation, which they implement under-the-hood by spawning multiple different windows. However, they don't send a "window created" event on the last window they create, thus AltTab can't see it.

Not sure what we can do about it. It's a real mess which originates from Books.app, with no workaround I can imagine for us to side-step the nonsense.

**Comment by JaiRelan** [comment]
Same issue here... Was having a problem with Bear app yesterday, today alt-tab decided that the Books app wasn't it's friend.

**Comment by lwouis** [comment]
@JaiRelan yes, both these apps have problems. I've added multiple work-arounds in AltTab, but essentially it's tape around a leak. Ideally these 2 apps would behave properly like other apps, and there wouldn't be a problem in the first place. However, especially for Books, it's unlikely to happen, so we are left imagining ever-more-complex workarounds to deal with the non-standard behavior of these apps.

**Comment by JaiRelan** [comment]
Of course... Totally get that. 

Thank you so much for working so hard to patch the bugs... Live Alt-Tab and please let me know if I can help with anything

**Comment by lwouis** [comment]
I tested today, and opening books was working fine. Closing this ticket. Please let me know if some scenario inside Books.app still doesn't work. Please open a new ticket if that's the case, with steps to reproduce and a video.
